exploration/number-selection.md

```diff
@@ -92,44 +95,343 @@ ICU MF1 messages using `plural` and `selectordinal` should be representable in M
+>
+> For example, if the value of the variable `num` were the string
+> `-1234.567`, it would behave identically to the local
+> variable in this example:
+> ```
+> .local $example = {|-1234.567| :number}
+> {{{$num :number} == {$example}}}
+> ```
 
-The default `plural` value is presumed to be the most common use case,
-and it affords the least bad fallback when used incorrectly:
-Using "plural" for "exact" still selects exactly matching cases,
-whereas using "exact" for "plural" will not select LDML category matches.
-This might not be noticeable in the source language,
+> [!NOTE]
+> Implementations are encouraged to provide support for compound types or data structures
+> that provide additional semantic meaning to the formatting of number-like values.
+> For example, in ICU4J, the type `com.ibm.icu.util.Measure` can be used to communicate
+> a value that include a unit
+> or the type `com.ibm.icu.util.CurrencyAmount` can be used to set the currency and related
+> options (such as the number of fraction digits).
+
+
+### Options
+
+The following options and their values are required in the default registry to be available on the 
+function `:number`:
+- `select`
+   -  `plural` (default)
+   -  `ordinal`
+   -  `exact`
+- `compactDisplay`
+   - `short` (default)
+   - `long`
+- `notation`
+   - `standard` (default)
+   - `scientific`
+   - `engineering`
+   - `compact`
+- `numberingSystem`
+   - valid [Unicode Number System Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeNumberSystemIdentifier)
+     (default is locale-specific)
+- `signDisplay`
+   -  `auto` (default)
+   -  `always`
+   -  `exceptZero`
+   -  `never`
+- `style`
+  - `decimal` (default)
+  - `percent` (see [Percent Style](#percent-style) below)
+- `useGrouping`
+  - `true` (default)
+  - `false`
+- `minimumIntegerDigits`
+  - (non-negative integer, default: `1`)
+- `minimumFractionDigits`
+  - (non-negative integer)
+- `maximumFractionDigits`
+  - (non-negative integer)
+- `minimumSignificantDigits`
+  - (non-negative integer)
+- `maximumSignificantDigits`
+  - (non-negative integer)
+
+The following options and their values are required in the default registry to be available on the 
+function `:integer`:
+- `select`
+   -  `plural` (default)
+   -  `ordinal`
+   -  `exact`
+- `compactDisplay`
+   - `short` (default)
+   - `long`
+- `numberingSystem`
+   - valid [Unicode Number System Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeNumberSystemIdentifier)
+     (default is locale-specific)
+- `signDisplay`
+   -  `auto` (default)
+   -  `always`
+   -  `exceptZero`
+   -  `never`
+- `style`
+  - `decimal` (default)
+  - `percent` (see [Percent Style](#percent-style) below)
+- `useGrouping`
+  - `true` (default)
+  - `false`
+- `minimumIntegerDigits`
+  - (non-negative integer, default: `1`)
+- `minimumSignificantDigits`
+  - (non-negative integer)
```

Comment:
Please remove `minimumSignificantDigits` from `:integer`; it serves no purpose other than to create confusion when adding trailing zeros which it seems we want to avoid in `:integer` mode.

Reply:
You went too far; I am actually okay with `maximumSignificantDigits` on `:integer`, just not `minimum*` because the only purpose that serves is to create trailing zeros which are not needed in `:integer` mode.

Reply:
Fixed.